splunk/resource_splunk_admin_saml_groups.go

```diff
@@ -74,6 +94,8 @@ func adminSAMLGroupsRead(d *schema.ResourceData, meta interface{}) error {
 	if entry == nil {
 		d.SetId("")
 		return nil
+	} else {
+		d.SetId(name)
```

Comment:
The legacy function didn't set the Id each read. I added this so that changes between the legacy and external clients work in both ways. The legacy client used the name, and the external client uses the actual URL. Adding the explicit `SetId` in `adminSAMLGroupsRead` makes it possible to migrate from the external client back to the internal client if needed.

Reply:
@micahkemp-splunk 
Just curious if you had thought only use the URN as Id and not the entire URL? (without the HTTP scheme and the IP). I think that might look better for readability purposes?

Reply:
`id` comes back from the REST API as a full URL, so it made sense to me to use it directly. It _is_ parsed to get the title and namespace user/app, but it's stored in URL form by the external client specifically because that's how the REST API returns it.